### PR TITLE
fix api/mock circular ref

### PR DESCRIPTION
**Pull request by ryanfkeepers**
No logic changes.  Just code movement/renaming.

---

#### Does this PR need a docs update or release note?

- [x] :no_entry: No

#### Type of change

- [x] :broom: Tech Debt/Cleanup

### CODE REVIEW

File: src/pkg/services/m365/api/pagers/pagers.go [hunk 1]
```diff
@@ -40,9 +40,9 @@ type NextPager[T any] interface {
 	NextPage() (items []T, reset, done bool)
 }
 
-type nextPage[T any] struct {
-	items []T
-	reset bool
+type NextPage[T any] struct {
+	Items []T
+	Reset bool
```
Comment (ashmrtn) [inline src/pkg/services/m365/api/pagers/pagers.go:45]: I only did some quick greps on the branch, but I don't think these fields need to be exported

File: src/pkg/services/m365/api/pagers/pagers.go [hunk 2]
```diff
@@ -51,59 +51,59 @@ type NextPageResulter[T any] interface {
...
+var _ NextPageResulter[any] = &NextPageResults[any]{}
 
-type nextPageResults[T any] struct {
-	pages chan nextPage[T]
-	du    DeltaUpdate
-	err   error
+type NextPageResults[T any] struct {
+	Pages chan NextPage[T]
+	DU    DeltaUpdate
+	Err   error
```
Comment (ashmrtn) [inline src/pkg/services/m365/api/pagers/pagers.go:59]: I don't think these fields really need to be exported either

File: src/pkg/services/m365/api/drive_pager.go [hunk 3]
```diff
@@ -204,20 +205,20 @@ func (c Drives) EnumerateDriveItemsDelta(
...
 	deltaPager := c.newDriveItemDeltaPager(
 		driveID,
 		prevDeltaLink,
 		cc)
 
-	npr := &nextPageResults[models.DriveItemable]{
-		pages: make(chan nextPage[models.DriveItemable]),
+	npr := &pagers.NextPageResults[models.DriveItemable]{
+		Pages: make(chan pagers.NextPage[models.DriveItemable]),
 	}
```
Comment (ashmrtn) [inline src/pkg/services/m365/api/drive_pager.go:216]: this is the only place `Pages` is accessed so it seems reasonable to just make an initializer function for the struct so we can keep the field unexported

**Review by ashmrtn — APPROVED** [review]
LGTM though would really like to see fields switched back to being unexported

File: src/pkg/services/m365/api/pagers/pagers.go (hunk 1 above)
Comment (ryanfkeepers) [reply src/pkg/services/m365/api/pagers/pagers.go]: done; was autopiloting a bit.  thanks for the check.

Comment (ashmrtn) [reply src/pkg/services/m365/api/pagers/pagers.go]: Please be more attentive going forward. No matter how the code came about, as the code author and author of the PR you're responsible for vouching for the general correctness and quality of the code. Expecting reviewers to check for simple quality things places a _huge_ burden in terms of review time and effort on them if you also want actionable and correct review comments. Just figuring out what could and couldn't be exported in these structs required me to either remember all the usages of these variables or clone the branch myself and search through the code using other tools

The bottom line is if the team at large keeps doing this either the review quality will go down (non-actionable or incorrect comments), the code quality will go down (more difficult to make changes going forward), or the reviewers are going to become overwhelmed
